src/extensions/mountainview/MountainView/ViewContainer.tsx

```diff
@@ -1,46 +1,117 @@
 
 
 type Props = {
     views: View[]
     onViewClosed: (v: View) => void
-    width?: number
+    onSetViewArea: (v: View, area: 'north' | 'south') => void
+    width: number
+    height: number
 }
 
-const tabBarHeight = 56
+const tabBarHeight = 60
 
-const ViewContainer: FunctionComponent<Props & SizeMeProps> = ({ children, views, onViewClosed, size, width }) => {
-    const [currentView, setCurrentView] = useState<View | null>(null)
+const ViewContainer: FunctionComponent<Props> = ({ children, views, onViewClosed, onSetViewArea, width, height }) => {
+    const [currentNorthView, setCurrentNorthView] = useState<View | null>(null)
+    const [currentSouthView, setCurrentSouthView] = useState<View | null>(null)
+    const [activeArea, setActiveArea] = useState<'north' | 'south'>('north')
     useEffect(() => {
         views.forEach(v => {
+            if (!v.area) v.area = activeArea
             if (v.activate) {
                 v.activate = false
-                setCurrentView(v)
+                if (v.area === 'north') {
+                    setCurrentNorthView(v)
+                    setActiveArea('north')
+                }
+                else if (v.area === 'south') {
+                    setCurrentSouthView(v)
+                    setActiveArea('south')
+                }
             }
         })
-    }, [views])
+    }, [views, activeArea])
+
+    const H0 = (height - tabBarHeight * 2) / 2
+    const areas: {[key: string]: {
+        tabBarStyle: React.CSSProperties,
+        divStyle: React.CSSProperties
+    }} = {
+        'north': {
+            tabBarStyle: { left: 0, top: 0, width: width, height: tabBarHeight },
+            divStyle: {left: 0, top: tabBarHeight, width: width, height: H0}
+        },
+        'south': {
+            tabBarStyle: { left: 0, top: tabBarHeight + H0, width: width, height: tabBarHeight },
+            divStyle: { left: 0, top: tabBarHeight * 2 + H0, width: width, height: H0 }
+        }
+    }
+
+    const handleClick = useCallback((e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        const y = eventToPos(e)[1]
+        const newActiveArea = (y < tabBarHeight + H0) ? 'north' : 'south'
+        if (newActiveArea !== activeArea) {
+            setActiveArea(newActiveArea)
+        }
+    }, [H0, activeArea, setActiveArea])
+
+    const handleDragOver = useCallback((event: React.DragEvent<HTMLDivElement>) => {
+        // the following is needed otherwise we can't get the drop. See: https://stackoverflow.com/questions/50230048/react-ondrop-is-not-firing
+        event.stopPropagation();
+        event.preventDefault();
+    }, [])
+
+    const handleDragDrop = useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        const y = eventToPos(e)[1]
+        const dropArea = (y < tabBarHeight + H0) ? 'north' : 'south'
+        const viewId = e.dataTransfer.getData('viewId')
+        if (viewId) {
+            const view = views.filter(v => v.viewId === viewId)[0]
```

Comment:
If we have a viewId then we are dropping the tab element associated with a view.